api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.py

```diff
@@ -0,0 +1,83 @@
+import base64
+import hashlib
+import hmac
+import logging
+import time
+import urllib.parse
+from typing import Any, Union
+
+import httpx
+
+from core.tools.entities.tool_entities import ToolInvokeMessage
+from core.tools.tool.builtin_tool import BuiltinTool
+
+
+class DingTalkGroupBotTool(BuiltinTool):
+    def _invoke(self, user_id: str, tool_parameters: dict[str, Any]
+                ) -> Union[ToolInvokeMessage, list[ToolInvokeMessage]]:
+        """
+            invoke tools
+            Dingtalk group chat api: https://open.dingtalk.com/document/orgapp/custom-robot-access
+        """
+        content = tool_parameters.get('content', '')
+        if not content:
+            return self.create_text_message('Invalid parameter content')
+
+        access_token = tool_parameters.get('access_token', '')
```

Comment:
And use this to get sk `self.runtime.credentials["access_token"]`

Reply:
Thanks. Resolved.